src/pypdfium2/_helpers/pageobjects.py

```diff
@@ -155,6 +157,84 @@ def transform(self, matrix):
             raise PdfiumError("Failed to transform pageobject with matrix.")
 
 
+class PdfTextObj (PdfObject):
+    """
+    """
+    
+    def __init__(self, *args, textpage=None, **kwargs):
+        if textpage is not None:
+            kwargs.update(page=textpage.page, pdf=textpage.page.pdf)
+        super().__init__(*args, **kwargs)
+        self.textpage = textpage
+    
+    def get_text(self):
+        """
+        """
+        bufsize = pdfium_c.FPDFTextObj_GetText(self, self.textpage, None, 0)
+        buffer = ctypes.create_string_buffer(bufsize)
+        buffer_ptr = ctypes.cast(buffer, ctypes.POINTER(pdfium_c.FPDF_WCHAR))
+        
+        out_bufsize = pdfium_c.FPDFTextObj_GetText(self, self.textpage, buffer_ptr, bufsize)
+        if out_bufsize == 0:
+            raise PdfiumError("Failed to get text from textobject.")
+        assert bufsize == out_bufsize
+        
+        return buffer.raw[:bufsize-2].decode("utf-16-le", errors="ignore")
+    
+    def get_font(self):
+        """
+        """
+        raw_font = pdfium_c.FPDFTextObj_GetFont(self)
+        # not owned by the caller
+        return PdfFont(raw_font, self)
+    
+    def get_font_size(self):
+        """
+        """
+        r_size = ctypes.c_float()
+        ok = pdfium_c.FPDFTextObj_GetFontSize(self, r_size)
+        if not ok:
+            raise PdfiumError("Failed to get font size.")
+        return r_size.value
+
+
+class PdfFont (pdfium_i.AutoCastable):
+    """
+    """
+    
+    def __init__(self, raw, parent=None):
+        self.raw = raw
+        self.parent = parent
+    
+    def get_name(self, which="base"):
+        """
+        """
+        if which == "base":
+            api = pdfium_c.FPDFFont_GetBaseFontName
+        elif which == "family":
+            api = pdfium_c.FPDFFont_GetFamilyName
+        else:
+            raise ValueError(f"{which!r}")
+        
+        bufsize = api(self, None, 0)
+        buffer = ctypes.create_string_buffer(bufsize)
+        
+        out_bufsize = api(self, buffer, bufsize)
+        if out_bufsize == 0:
+            raise PdfiumError(f"Failed to get font {which} name.")
```

Comment:
todo: check for this case after the first call, and return None (or an empty string?) instead of raising error